rest-service/api/src/main/java/org/veupathdb/service/multiblast/service/http/JobService.java

```diff
@@ -200,13 +201,23 @@ public ReportWrap getReport(
   ) {
     log.trace("#getReport(jobID={}, format={}, zip={}, fields={}, maxDlSize={})", jobID, format, zip, fields, maxDlSize);
 
-    try {
-      if (!Format.isHex(jobID))
-        throw new NotFoundException();
+    if (!Format.isHex(jobID))
+      throw new NotFoundException();
 
-      FullUserJobRow job;
+    var jobHash = Format.hexToBytes(jobID);
+
+    try {
+      FullJobRow job;
+      UserRow    user;
       try (var db = new JobDBManager()) {
-        job = db.getUserJob(Format.hexToBytes(jobID), userId).orElseThrow(NotFoundException::new);
+        job = db.getJob(jobHash).orElseThrow(NotFoundException::new);
+
+        if (db.userIsLinkedToJob(userId, jobHash)) {
+          user = db.getUser(userId, jobHash).orElseThrow(InternalServerErrorException::new);
+        } else {
+          user = new UserRowImpl(jobHash, userId, null, null, true);
+          db.linkUserToJob(user);
+        }
```

Comment:
Instead of getting the job for the logged-in user, now we do:
1. Load job from the DB
2. Check if the requester is attached to the job
3. Optionally attach the requester to the job.